### PR TITLE
Added RKE module

### PR DESCRIPTION
**Pull request by suryatejaboorlu**

### CODE REVIEW

File: aws/ami.tf [hunk 1]
```diff
@@ -1,14 +1,15 @@
 data "aws_ami" "ubuntu" {
   most_recent = true
-  owners      = ["099720109477"] # Canonical
 
   filter {
     name   = "name"
-    values = ["ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-*"]
+    values = ["ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*"]
```
Comment (leodotcloud) [inline aws/ami.tf:6]: Do we need this? Aren't things working on 22.04?

File: aws/keys.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+resource "tls_private_key" "node-key" {
+  algorithm = "RSA"
+}
+
+resource "aws_key_pair" "rke-node-key" {
+  key_name   = "surya-test-keypair"
```
Comment (leodotcloud) [inline aws/keys.tf:6]: Plz remove this hardcoding, use a variable.

File: aws/nodes.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
+locals {
+  cluster_id_tag = {
```
Comment (leodotcloud) [inline aws/nodes.tf:2]: Please add comments on what's this for?

File: aws/nodes.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = local.cluster_id_tag
+}
+
+resource "aws_instance" "rke-node" {
+  count = 4
```
Comment (leodotcloud) [inline aws/nodes.tf:38]: Hard coding here as well.

File: aws/nodes.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = local.cluster_id_tag
+}
+
+resource "aws_instance" "rke-node" {
```
Comment (leodotcloud) [inline aws/nodes.tf:37]: `rke-node`: Plz ensure the naming conventions are similar to other folders.

File: aws/nodes.tf [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  count = 4
+
+  ami                    = data.aws_ami.ubuntu.id
+  instance_type          = var.instance_type
+  key_name               = aws_key_pair.rke-node-key.id
+  iam_instance_profile   = "rk-all-role"
```
Comment (leodotcloud) [inline aws/nodes.tf:43]: Do we really need an IAM instance profile?

File: aws/nodes.tf [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  vpc_security_group_ids = [aws_security_group.allow-all.id]
+  tags                   = local.cluster_id_tag
+
+  provisioner "remote-exec" {
+    connection {
+      host        = coalesce(self.public_ip, self.private_ip)
+      type        = "ssh"
+      user        = "ubuntu"
```
Comment (leodotcloud) [inline aws/nodes.tf:51]: Hard coding.

File: aws/nodes.tf [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      private_key = tls_private_key.node-key.private_key_pem
+    }
+
+    inline = [
+      "curl ${var.docker_install_url} | sh",
```
Comment (leodotcloud) [inline aws/nodes.tf:56]: Plz ensure this is how we are installing docker in the other module as well.

File: aws/nodes.tf [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      "sudo usermod -a -G docker ubuntu",
```
Comment (leodotcloud) [inline aws/nodes.tf:57]: Hardcoding.

File: aws/output.tf [hunk 10]
```diff
@@ -0,0 +1,16 @@
+output "private_key" {
+  value = tls_private_key.node-key.private_key_pem
+  sensitive = true
+}
+
+output "ssh_username" {
+  value = "ubuntu"
+}
+
+output "addresses" {
```
Comment (leodotcloud) [inline aws/output.tf:10]: Same for the outputs too. Plz follow the same convention as other module.

**Review by leodotcloud — CHANGES_REQUESTED** [review]

**Comment by suryatejaboorlu** [comment]
Closing as this code is not up to our requirement.

**Pull request closed without merging**